src/functions/awesome_memgraph_functions.hpp

```diff
@@ -0,0 +1,1465 @@
+    if (s1.size() < s2.size()) return false;
+    return std::equal(s2.begin(), s2.end(), s1.begin());
+  }
+};
+
+template <typename TypedValueT, typename FunctionContextT>
+inline const auto StartsWith = StringMatchOperator<StartsWithPredicate<TypedValueT>, TypedValueT, FunctionContextT>;
+
+// Check if s1 ends with s2.
+template <typename TypedValueT>
+struct EndsWithPredicate {
+  static constexpr const char *name = "endsWith";
+  bool operator()(const typename TypedValueT::TString &s1, const typename TypedValueT::TString &s2) const {
+    if (s1.size() < s2.size()) return false;
+    return std::equal(s2.rbegin(), s2.rend(), s1.rbegin());
+  }
+};
+
+template <typename TypedValueT, typename FunctionContextT>
+inline const auto EndsWith = StringMatchOperator<EndsWithPredicate<TypedValueT>, TypedValueT, FunctionContextT>;
+
+// Check if s1 contains s2.
+template <typename TypedValueT>
+struct ContainsPredicate {
+  static constexpr const char *name = "contains";
+  bool operator()(const typename TypedValueT::TString &s1, const typename TypedValueT::TString &s2) const {
+    if (s1.size() < s2.size()) return false;
+    return s1.find(s2) != std::string::npos;
+  }
+};
+
+template <typename TypedValueT, typename FunctionContextT>
+inline const auto Contains = StringMatchOperator<ContainsPredicate<TypedValueT>, TypedValueT, FunctionContextT>;
+
+template <typename TypedValueT, typename FunctionContextT>
+TypedValueT Assert(const TypedValueT *args, int64_t nargs, const FunctionContextT &ctx) {
+  FType<TypedValueT, Bool, Optional<String>>("assert", args, nargs);
+  if (!args[0].ValueBool()) {
+    std::string message("Assertion failed");
+    if (nargs == 2) {
+      message += ": ";
+      message += args[1].ValueString();
+    }
+    message += ".";
+    throw FunctionRuntimeException(message);
+  }
+  return TypedValueT(args[0], ctx.memory);
+}
+
+template <typename TypedValueT, typename FunctionContextT>
+TypedValueT Counter(const TypedValueT *args, int64_t nargs, const FunctionContextT &context) {
+  FType<TypedValueT, String, Integer, Optional<NonZeroInteger>>("counter", args, nargs);
+  int64_t step = 1;
+  if (nargs == 3) {
+    step = args[2].ValueInt();
+  }
+
+  auto [it, inserted] = context.counters->emplace(args[0].ValueString(), args[1].ValueInt());
+  auto value = it->second;
+  it->second += step;
+
+  return TypedValueT(value, context.memory);
+}
+
+template <typename TypedValueT, typename FunctionContextT>
+TypedValueT Id(const TypedValueT *args, int64_t nargs, const FunctionContextT &ctx) {
+  FType<TypedValueT, Or<Null, Vertex, Edge>>("id", args, nargs);
+  const auto &arg = args[0];
+  if (arg.IsNull()) {
+    return TypedValueT(ctx.memory);
+  }
+  if (arg.IsVertex()) {
+    return TypedValueT(static_cast<int64_t>(arg.ValueVertex().CypherId()), ctx.memory);
+  }
+  return TypedValueT(static_cast<int64_t>(arg.ValueEdge().CypherId()), ctx.memory);
+}
```

Comment:
I think the current solution is not good, because it adds a dummy function to both accessors. I think for both accessors there are better solutions.
For vertices:
1. Remove it completely. This was necessary to make `ScanById` possible, so targeting a single vertex. In the distributed case the same can be achieved with a `MATCH`. The users has to "unfold" the properties, but that is something we can live for now.
2. Make it return a map that is in a format of `{primary_label: "PrimaryLabelName", key: ["pk_value1", "pk_value2", 4, 5]}`. This way the same semantic can be kept as before.

For edges I think we should use the `Gid` as before.